src/ngx_http_lua_pipe.c

```diff
@@ -0,0 +1,2478 @@
+#define REASON_UNKNOWN_CODE  3
+
+
+void
+ngx_http_lua_pipe_init(void)
+{
+    ngx_rbtree_init(&ngx_http_lua_pipe_rbtree,
+                    &ngx_http_lua_pipe_proc_sentinel, ngx_rbtree_insert_value);
+}
+
+
+ngx_int_t
+ngx_http_lua_pipe_add_signal_handler(ngx_cycle_t *cycle)
+{
+    ngx_event_t         *rev;
+#if (NGX_HTTP_LUA_HAVE_SIGNALFD)
+    sigset_t             set;
+
+#else
+    int                  rc;
+    struct sigaction     sa;
+#endif
+
+#if (NGX_HTTP_LUA_HAVE_SIGNALFD)
+    if (sigemptyset(&set) != 0) {
+        ngx_log_error(NGX_LOG_ERR, cycle->log, ngx_errno,
+                      "lua pipe init signal set failed");
+        return NGX_ERROR;
+    }
+
+    if (sigaddset(&set, SIGCHLD) != 0) {
+        ngx_log_error(NGX_LOG_ERR, cycle->log, ngx_errno,
+                      "lua pipe add SIGCHLD to signal set failed");
+        return NGX_ERROR;
+    }
+
+    if (sigprocmask(SIG_BLOCK, &set, NULL) != 0) {
+        ngx_log_error(NGX_LOG_ERR, cycle->log, ngx_errno,
+                      "lua pipe block SIGCHLD failed");
+        return NGX_ERROR;
+    }
+
+    ngx_http_lua_signalfd = signalfd(-1, &set, SFD_NONBLOCK|SFD_CLOEXEC);
+    if (ngx_http_lua_signalfd < 0) {
+        ngx_log_error(NGX_LOG_ERR, cycle->log, ngx_errno,
+                      "lua pipe create signalfd instance failed");
+        return NGX_ERROR;
+    }
+
+#else /* !(NGX_HTTP_LUA_HAVE_SIGNALFD) */
+#   if (NGX_HTTP_LUA_HAVE_PIPE2)
+    rc = pipe2(ngx_http_lua_sigchldfd, O_NONBLOCK|O_CLOEXEC);
+#   else
+    rc = pipe(ngx_http_lua_sigchldfd);
+#   endif
+
+    if (rc == -1) {
+        ngx_log_error(NGX_LOG_ERR, cycle->log, ngx_errno,
+                      "lua pipe init SIGCHLD fd failed");
+        return NGX_ERROR;
+    }
+
+#   if !(NGX_HTTP_LUA_HAVE_PIPE2)
+    if (ngx_nonblocking(ngx_http_lua_read_sigfd) == -1) {
+        ngx_log_error(NGX_LOG_ERR, cycle->log, ngx_errno,
+                      "lua pipe " ngx_nonblocking_n " SIGCHLD read fd failed");
+        goto failed;
+    }
+
+    if (ngx_nonblocking(ngx_http_lua_write_sigfd) == -1) {
+        ngx_log_error(NGX_LOG_ERR, cycle->log, ngx_errno,
+                      "lua pipe " ngx_nonblocking_n " SIGCHLD write fd failed");
+        goto failed;
+    }
+
+    /* it's ok not to set the pipe fd with O_CLOEXEC. This requires
+     * extra syscall */
+#   endif /* !(NGX_HTTP_LUA_HAVE_PIPE2) */
+#endif /* NGX_HTTP_LUA_HAVE_SIGNALFD */
+
+    ngx_http_lua_sigfd_conn = ngx_get_connection(ngx_http_lua_read_sigfd,
+                                                 cycle->log);
+    if (ngx_http_lua_sigfd_conn == NULL) {
+        goto failed;
+    }
+
+    ngx_http_lua_sigfd_conn->log = cycle->log;
+    ngx_http_lua_sigfd_conn->recv = ngx_http_lua_pipe_fd_read;
+    rev = ngx_http_lua_sigfd_conn->read;
+    rev->log = ngx_http_lua_sigfd_conn->log;
+    rev->handler = ngx_http_lua_pipe_sigchld_event_handler;
+
+#ifdef HAVE_SOCKET_CLOEXEC_PATCH
+    rev->skip_socket_leak_check = 1;
```

Comment:
What is this struct member? I cannot compile this branch in my environment (where I apply the socket_cloexec patch) because of this:

```
lua-nginx-module/src/ngx_http_lua_pipe.c:1460:12: error: ‘ngx_event_t’ {aka ‘struct ngx_event_s’} has no member named ‘skip_socket_leak_check’
     c->read->skip_socket_leak_check = 1;
            ^~
make[1]: *** [objs/Makefile:2540: objs/addon/src/ngx_http_lua_pipe.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory 'lua-nginx-module/buildroot/nginx-1.13.6'
make: *** [Makefile:8: build] Error 2
```

The socket_cloexec patch does not seem to have any mention of it: https://github.com/openresty/openresty/blob/master/patches/nginx-1.13.6-socket_cloexec.patch

Reply:
Err, you need to update the `socket_cloexec.patch` from this sister PR: https://github.com/openresty/openresty/pull/412